net: lib: nrf_cloud: Clean up KConfigs so they don't clutter unrelated samples (#20074)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_3.0.rst

This section describes the changes related to libraries.

nRF Cloud library
-----------------

.. toggle::

   For applications and samples using the :ref:`lib_nrf_cloud` library:

>    * It will be required to enable :kconfig:option: `CONFIG_NRF_CLOUD` Kconfig option in order to access the nRF Cloud libraries.

Reviewer: ```suggestion
   * It will be required to enable :kconfig:option:`CONFIG_NRF_CLOUD` Kconfig option in order to access the nRF Cloud libraries.
```

>    * It will be required to enable :kconfig:option:`CONFIG_NRF_CLOUD` Kconfig option in order to access the nRF Cloud libraries.
>      This option is now disabled by default to prevent the unintended inclusion of nRF Cloud Kconfig variables in non-nRF Cloud projects, addressing a previous issue.

Reviewer 2: ```suggestion
   * You must set the :kconfig:option:`CONFIG_NRF_CLOUD` Kconfig option to access the nRF Cloud libraries.
     This option is now disabled by default to prevent the unintended inclusion of nRF Cloud Kconfig variables in non-nRF Cloud projects, addressing a previous issue.
```
